packages/sanity/src/core/store/_legacy/document/document-store.ts

```diff
@@ -33,12 +33,20 @@ import {type IdPair} from './types'
  * @beta */
 export type QueryParams = Record<string, string | number | boolean | string[]>
 
-function getIdPairFromPublished(publishedId: string): IdPair {
+function getIdPairFromPublished(publishedId: string, version?: string): IdPair {
   if (isDraftId(publishedId)) {
     throw new Error('editOpsOf does not expect a draft id.')
   }
 
-  return {publishedId, draftIds: [getDraftId(publishedId)]}
+  return {
+    publishedId,
+    draftId: getDraftId(publishedId),
+    ...(version
+      ? {
+          versionId: [version, publishedId].join('.'),
+        }
+      : {}),
+  }
```

Comment:
This feels like we are doing the same change in two places, and could easily be missed.  https://github.com/sanity-io/sanity/pull/7251/files#diff-cdd1fc4337b45f431507fa1b2ebb702510d7c9472df372033daaa9f1871add8c
What do you think if we replace this for 
```suggestion
   return getIdPair(publishedId, {version})
  }
```

Reply:
Oh yeah! That looks like a really nice improvement to me! Thanks.